src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -504,6 +504,24 @@ internal static class Assembly
                 .CreateDelegate<Func<System.Reflection.Assembly, string, bool, bool, Type>>();
         }
 
+        internal static class AssemblyName
+        {
+            private const string TypeName = "System.Reflection.AssemblyName";
+
+            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            private static readonly PropertyInfo CodeBase = Type
+                .GetTypeInfo()
+                .GetDeclaredProperty(nameof(CodeBase));
```

Comment:
`CodeBase` is not part of the contract API for CoreFx at https://github.com/dotnet/corefx/blob/master/src/System.Reflection/ref/System.Reflection.cs.  Is it part of whatever implementation that forwards to, such that this will work on CoreCLR?

Our normal pattern for PortableShim is that it's only things that actually supported on both Desktop and CoreCLR, just not included in the portable profile we're currently targeting.

Tagging @tmat, @agocke, and @jaredpar to see if they have suggestions here.